src/bin/tikv-ctl.rs

```diff
@@ -21,25 +21,381 @@ extern crate clap;
 use tikv::util::{self, escape, unescape};
-use tikv::util::codec::bytes::encode_bytes;
-use tikv::raftstore::store::keys;
-use tikv::raftstore::store::engine::{IterOption, Iterable, Peekable};
-use tikv::storage::{CfName, ALL_CFS, CF_DEFAULT, CF_LOCK, CF_RAFT, CF_WRITE};
-use tikv::storage::mvcc::{Lock, Write};
-use tikv::storage::types::Key;
+use tikv::raftstore::store::{keys, Engines};
+use tikv::raftstore::store::debug::{Debugger, RegionInfo};
+use tikv::storage::{ALL_CFS, CF_DEFAULT, CF_LOCK, CF_RAFT, CF_WRITE};
+
+enum DebugExecutor {
+    Remote(DebugClient),
+    Local(Debugger),
+}
+
+impl DebugExecutor {
+    fn from_args(matches: &ArgMatches) -> Self {
+        let remote = matches.value_of("host");
+        let local = matches.value_of("db");
+        match (remote, local) {
+            (Some(_), Some(_)) => {
+                eprintln!(r#""host" and "db" can not be passed together!"#);
+                process::exit(1);
+            }
+            (None, None) => {
+                eprintln!(r#"please pass "host" or "db""#);
+                process::exit(1);
+            }
+            (None, Some(path)) => {
+                let db = Arc::new(util::rocksdb::open(path, ALL_CFS).unwrap());
+                let raft_db = if let Some(raftdb_path) = matches.value_of("raftdb") {
+                    Arc::new(util::rocksdb::open(raftdb_path, &[CF_DEFAULT]).unwrap())
+                } else {
+                    let raftdb_path = PathBuf::from(path).join("../raft");
+                    Arc::new(
+                        util::rocksdb::open(raftdb_path.to_str().unwrap(), &[CF_DEFAULT]).unwrap(),
+                    )
+                };
+                DebugExecutor::Local(Debugger::new(Engines::new(db, raft_db)))
+            }
+            (Some(remote), None) => {
+                let env = Arc::new(Environment::new(1));
+                let channel = ChannelBuilder::new(env).connect(remote);
+                let client = DebugClient::new(channel);
+                DebugExecutor::Remote(client)
+            }
+        }
+    }
+
+    fn print(&self, cf: &str, key: &str) {
+        let db = DBType::KV;
+        let key = unescape(key);
+        let value = match *self {
+            DebugExecutor::Remote(ref client) => {
+                let mut get_req = GetRequest::new();
+                get_req.set_db(db);
+                get_req.set_cf(cf.to_owned());
+                get_req.set_key(key);
+                client
+                    .get(get_req)
+                    .map(|resp| escape(resp.get_value()))
+                    .unwrap_or_else(Self::report_and_exit)
+            }
+            DebugExecutor::Local(ref debugger) => debugger
+                .get(db, cf, &key)
+                .map(|bytes| escape(&bytes))
+                .unwrap_or_else(Self::report_and_exit),
+        };
+        println!("value: {}", value);
+    }
+
+    fn region_size(&self, region_id: u64, cfs: Vec<&str>) {
```

Comment:
Weird name. It feels like it's going to return the region size. Maybe `verb+noun` is a better choice.